### ISSUE TITLE
S3's `createMultipartUpload(input:)` and `completeMultipartUpload(input:)` both fail for uploads to the protected and private folders 

### ISSUE THREAD

**Issue by ruisebas** [awslabs/aws-sdk-swift#615]
## Description
Attempting to being and complete a multipart upload by calling `S3Client.createMultipartUpload(input:)` and `S3Client.completeMultipartUpload(input:)` for objects uploaded to the user's `protected` and `private` folder throw error and results in no object being successfully uploaded.

It works for uploading to the `public` folder though.

## Environment
* AWS Swift SDK version used: 0.2.6
* Operating System and version (iOS/macOS/Linux): iOS
* Xcode version (if applicable): 13.4.1

## Reproduction Steps
Having the following structure in an S3 bucket:

```
aBucket
├── public
│   └── existingFile
│ 
├── protected
│   └── region:userId1
│       └── existingFile
│ 
├── private
│   └── region:userId1
│       └── existingFile
│
```

This code that calls the `createMultipartUpload` API:
```swift
let bucket = // The bucket, i.e. "aBucket"
let key = // The key for the multipart upload, i.e. "public/newFile", "protected/region:user1/newFile" or "private/region:user1/newFile"
let input = CreateMultipartUploadInput(bucket: bucket,
                                       key: key)

do {
    let response = try await awsS3.createMultipartUpload(input: input)
    // Handle response...
} catch {
    // Handle error...
}

```

And this code that calls the `completeMultipartUpload` API:

```swift
let bucket = // The bucket, i.e. "aBucket"
let key = // The key for the multipart upload, i.e. "public/newFile", "protected/region:user1/newFile" or "private/region:user1/newFile"
let uploadId = // The UploadID
let completedMultipartUpload = // A CompletedMultipartUpload object containing the parts

let input = CompleteMultipartUploadInput(bucket: bucket, key: key, multipartUpload: completedMultipartUpload, uploadId: uploadId)
do {
    let response = try await awsS3.completeMultipartUpload(input: input)
    // Handle response...
} catch {
    // Handle error...
}
```

## Observed Behavior

When using `key: "public/newFile"`, both requests are successful and the object is uploaded. 

However, attempting to use `key: "protected/region:user1/newFile"` and `key: "private/region:user1/newFile"` results in the requests failing with the following errors:

**CreateMultipartUpload**
<details>
<summary><tt>The request signature we calculated does not match the signature you provided</tt>.<br><tt>Check your key and signing method.</tt> <i>(Click to see the full error)</i></summary>

```
▿ SdkError<CreateMultipartUploadOutputError>
  ▿ client : 2 elements
    ▿ .0 : ClientError
      ▿ retryError : SdkError<CreateMultipartUploadOutputError>
        ▿ service : 2 elements
          ▿ .0 : CreateMultipartUploadOutputError
            ▿ unknown : UnknownAWSHttpServiceError
              - _isThrottling : false
              ▿ _statusCode : Optional<HttpStatusCode>
                - some : http_status_403
              ▿ _headers : Optional<Headers>
                ▿ some : x-amz-request-id: «REDACTED», 
Content-Type: application/xml, 
Date: Mon, 12 Sep 2022 18:11:33 GMT, 
Transfer-Encoding: chunked, 
x-amz-id-2: «REDACTED», 
Server: AmazonS3
                  ▿ headers : 6 elements
                    ▿ 0 : Header
                      - name : "x-amz-request-id"
                      ▿ value : 1 element
                        - 0 : "«REDACTED»"
                    ▿ 1 : Header
                      - name : "x-amz-id-2"
                      ▿ value : 1 element
                        - 0 : "«REDACTED»"
                    ▿ 2 : Header
                      - name : "Content-Type"
                      ▿ value : 1 element
                        - 0 : "application/xml"
                    ▿ 3 : Header
                      - name : "Transfer-Encoding"
                      ▿ value : 1 element
                        - 0 : "chunked"
                    ▿ 4 : Header
                      - name : "Date"
                      ▿ value : 1 element
                        - 0 : "Mon, 12 Sep 2022 18:11:33 GMT"
                    ▿ 5 : Header
                      - name : "Server"
                      ▿ value : 1 element
                        - 0 : "AmazonS3"
              ▿ _message : Optional<String>
                - some : "The request signature we calculated does not match the signature you provided. Check your key and signing method."
              ▿ _requestID : Optional<String>
                - some : "«REDACTED»"
              - _retryable : false
              - _type : ClientRuntime.ErrorType.unknown
          ▿ .1 : 
Status Code: http_status_403 
 x-amz-request-id: «REDACTED», 
Content-Type: application/xml, 
Date: Mon, 12 Sep 2022 18:11:33 GMT, 
Transfer-Encoding: chunked, 
x-amz-id-2: «REDACTED», 
Server: AmazonS3
    - .1 : nil
```
</details>

**CompleteMultipartUpload**
<details>
<summary><tt>The request signature we calculated does not match the signature you provided</tt>.<br><tt>Check your key and signing method.</tt> <i>(Click to see the full error)</i></summary>

```
  ▿ SdkError<CompleteMultipartUploadOutputError>
  ▿ client : 2 elements
    ▿ .0 : ClientError
      ▿ retryError : SdkError<CompleteMultipartUploadOutputError>
        ▿ service : 2 elements
          ▿ .0 : CompleteMultipartUploadOutputError
            ▿ unknown : UnknownAWSHttpServiceError
              - _isThrottling : false
              ▿ _statusCode : Optional<HttpStatusCode>
                - some : http_status_403
              ▿ _headers : Optional<Headers>
                ▿ some : Connection: close, 
x-amz-id-2: «REDACTED», 
Transfer-Encoding: chunked, 
Server: AmazonS3, 
x-amz-request-id: «REDACTED», 
Content-Type: application/xml, 
Date: Mon, 12 Sep 2022 17:43:34 GMT
                  ▿ headers : 7 elements
                    ▿ 0 : Header
                      - name : "x-amz-request-id"
                      ▿ value : 1 element
                        - 0 : "«REDACTED»"
                    ▿ 1 : Header
                      - name : "x-amz-id-2"
                      ▿ value : 1 element
                        - 0 : "«REDACTED»"
                    ▿ 2 : Header
                      - name : "Content-Type"
                      ▿ value : 1 element
                        - 0 : "application/xml"
                    ▿ 3 : Header
                      - name : "Transfer-Encoding"
                      ▿ value : 1 element
                        - 0 : "chunked"
                    ▿ 4 : Header
                      - name : "Date"
                      ▿ value : 1 element
                        - 0 : "Mon, 12 Sep 2022 17:43:34 GMT"
                    ▿ 5 : Header
                      - name : "Server"
                      ▿ value : 1 element
                        - 0 : "AmazonS3"
                    ▿ 6 : Header
                      - name : "Connection"
                      ▿ value : 1 element
                        - 0 : "close"
              ▿ _message : Optional<String>
                - some : "The request signature we calculated does not match the signature you provided. Check your key and signing method."
              ▿ _requestID : Optional<String>
                - some : "«REDACTED»"
              - _retryable : false
              - _type : ClientRuntime.ErrorType.unknown
          ▿ .1 : 
Status Code: http_status_403 
 Connection: close, 
x-amz-id-2: «REDACTED», 
Transfer-Encoding: chunked, 
Server: AmazonS3, 
x-amz-request-id: «REDACTED», 
Content-Type: application/xml, 
Date: Mon, 12 Sep 2022 17:43:34 GMT
    - .1 : nil
```
</details>

## Expected Behavior
The requests should be able to upload objects to the `protected` and `private` folders as well.

## Possible Solution
I believe it's the same issue as https://github.com/awslabs/aws-sdk-swift/issues/613. 

Both generated methods have the `SigV4Config` object created with a `useDoubleURIEncode: false` flag:
```swift
// ...
let sigv4Config = AWSClientRuntime.SigV4Config(useDoubleURIEncode: false, shouldNormalizeURIPath: false, signedBodyHeader: .contentSha256, unsignedBody: false)
// ...
```

If I change it to `useDoubleURIEncode: true` (by duplicating the code in a local method instead), then both requests work for all three keys and successfully upload the objects.

**Comment by jbelkins** [comment]
I think this issue is due to the colon character in your bucket key, which the SDK is not properly encoding prior to making the HTTP request.  A fix is forthcoming, and should ship in the next release.

**Comment by jbelkins** [comment]
Resolved in release [0.3.0](https://github.com/awslabs/aws-sdk-swift/releases/tag/0.3.0).  Closing.
